osu.Game/Screens/OnlinePlay/Multiplayer/Match/MultiplayerReadyButton.cs

```diff
@@ -123,47 +172,43 @@ private void endOperation()
 
         private void updateState()
         {
-            var localUser = Client.LocalUser;
-
-            int newCountReady = Room?.Users.Count(u => u.State == MultiplayerUserState.Ready) ?? 0;
-            int newCountTotal = Room?.Users.Count(u => u.State != MultiplayerUserState.Spectating) ?? 0;
-
-            switch (localUser?.State)
+            if (Room == null)
             {
-                default:
-                    button.Text = "Ready";
-                    updateButtonColour(true);
-                    break;
+                enabled.Value = false;
+                return;
+            }
 
-                case MultiplayerUserState.Spectating:
-                case MultiplayerUserState.Ready:
-                    string countText = $"({newCountReady} / {newCountTotal} ready)";
+            var localUser = Client.LocalUser;
 
-                    if (Room?.Host?.Equals(localUser) == true)
-                    {
-                        button.Text = $"Start match {countText}";
-                        updateButtonColour(true);
-                    }
-                    else
-                    {
-                        button.Text = $"Waiting for host... {countText}";
-                        updateButtonColour(false);
-                    }
+            int newCountReady = Room.Users.Count(u => u.State == MultiplayerUserState.Ready);
+            int newCountTotal = Room.Users.Count(u => u.State != MultiplayerUserState.Spectating);
 
-                    break;
+            if (Room.Countdown != null)
+                countdownButton.Alpha = 0;
+            else
+            {
+                switch (localUser?.State)
+                {
+                    default:
+                        countdownButton.Alpha = 0;
```

Comment:
It can't be moved to the countdown button because room updates are scheduled. Unless I add hacks such as `AlwaysPresent`/`IsPresent` overrides... But we've previously decided that components shouldn't be transforming themselves in general, so the way it currently is feels better in that regard.